From 2e95b29fe3254b6d0234bb63470969c44d2997a1 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Mon, 4 Mar 2024 09:41:03 +0100
Subject: [PATCH 0770/1695] dt-bindings: update device tree bindings
 documentation

Add HSE / HSI input clocks.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 3abe28293b10f8d5359ec6537ef0ebe1080b0d78 dt-bindings: update device tree bindings documentation]
Change-Id: I2765b53432374f0f16c17db0231abb8709ccbe07
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/364246
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/374137
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 .../bindings/hwmon/st,stm32-hse-monitoring.yaml    | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/documentation/devicetree/bindings/hwmon/st,stm32-hse-monitoring.yaml b/documentation/devicetree/bindings/hwmon/st,stm32-hse-monitoring.yaml
index 107e4a941..64c971444 100644
--- a/documentation/devicetree/bindings/hwmon/st,stm32-hse-monitoring.yaml
+++ b/documentation/devicetree/bindings/hwmon/st,stm32-hse-monitoring.yaml
@@ -13,7 +13,8 @@ description: |
   STMicroelectronics's STM32 MPUs are able to monitor High Speed External
   oscillator frequency. Based on a lptimer, it is possible to detect a
   overclock frequency which can corrupt the system. The frequency overclock
-  detection is detected thanks to High Speed Internal oscillator reference.
+  detection is detected thanks to High Speed Internal oscillator reference
+  (hsi).
 
 properties:
   compatible:
@@ -23,6 +24,15 @@ properties:
     description: Counter used to monitor the frequency.
       Refer to counter.yaml for more details.
 
+  clocks:
+    minItems: 2
+    maxItems: 2
+
+  clock-names:
+    items:
+      - const: hse
+      - const: hsi
+
 required:
   - compatible
 
@@ -33,6 +43,8 @@ examples:
   hse_monitor: hse-monitor {
     compatible = "st,freq-monitor";
     counter = <&lptimer3 1 1 0 0>;
+    clocks = <&rcc CK_HSE>, <&rcc CK_HSI>;
+    clock-names = "hse", "hsi";
   };
 
 ...
-- 
2.47.2

